replica/compaction_group.hh

```diff
@@ -233,23 +234,25 @@ protected:
     // The list entries are unlinked automatically when the storage group, they belong to, is removed.
     compaction_group_list _compaction_groups;
     storage_group_map _storage_groups;
-
+    // Prevents _storage_groups from having its elements inserted or deleted while other layer iterates
+    // over them (or over _compaction_groups).
+    seastar::rwlock _lock;
 public:
     virtual ~storage_group_manager();
 
+    seastar::rwlock& get_rwlock() noexcept {
+        return _lock;
+    }
+
```

Comment:
It's better not to expose the lock, and instead only expose functions that use the lock safely, like for_each_storage_group. Exposing the lock publicly makes the caller responsible for understanding exactly what it locks.  Please follow up with removal.